### PR TITLE
Remove rules for Array, vect, circshift, fill, reverse -- ChainRules 1.5

### PR DESCRIPTION
**Pull request by willtebbutt**
New rules added in CR-1.5 should make the rules removed here redundant. I've left the corresponding tests to ensure nothing breaks.

### CODE REVIEW

**Comment by willtebbutt** [comment]
GPU failure doesn't seem to be related to this PR.

**Comment by DhairyaLGandhi** [comment]
Pretty weird that 1.5 errors. 

Could you link to the rules in CR? Mostly lgtm.

**Comment by willtebbutt** [comment]
> Could you link to the rules in CR?

Where / in what format would you like me to do this?

**Comment by mcabbott** [comment]
Could remove most of the `cat` rules too, although not sure this one test will pass on CR's rules, so `vcat` may have to stay:

https://github.com/FluxML/Zygote.jl/blob/master/test/cuda.jl#L105

Could also remove `repeat`, the rules here appear to be in CR:

https://github.com/JuliaDiff/ChainRules.jl/blob/master/src/rulesets/Base/array.jl#L63

**Comment by willtebbutt** [comment]
@mcabbott I would prefer to keep the scope of this PR to the changes introduced in CR 1.5 -- would you mind if we made those other changes in a different PR?

**Comment by mcabbott** [comment]
Sure, no obligation to do more.

File: Project.toml [hunk 1]
```diff
@@ -23,7 +23,7 @@ ZygoteRules = "700de1a5-db45-46bc-99cf-38207098b444"
 
 [compat]
 AbstractFFTs = "0.5, 1.0"
-ChainRules = "1"
+ChainRules = "1.5"
 ChainRulesCore = "1.0.1"
```
Comment (mcabbott) [inline Project.toml:27]: Probably doesn't matter, but CR 1.5 needs CRC 1.1
```suggestion
ChainRulesCore = "1.1"
```

**Review by mcabbott — APPROVED** [review]

File: src/lib/array.jl [hunk 2]
```diff
@@ -3,22 +3,17 @@ using FillArrays: AbstractFill, getindex_value
...
 
-@adjoint Base.vect(xs...) = Base.vect(xs...), Δ -> (Δ...,)
-
 @adjoint copy(x::AbstractArray) = copy(x), ȳ -> (ȳ,)
 
 @adjoint collect(x::Tuple) = collect(x), dy -> (Tuple(dy),)
 @adjoint collect(x::AbstractArray) = collect(x), dy -> (dy,)
 
 # Array Constructors
-@adjoint (::Type{T})(x::T) where T<:Array = T(x), ȳ -> (ȳ,)
```
Comment (DhairyaLGandhi) [inline src/lib/array.jl:21]: Do we have an equivalent in CR?

File: src/lib/array.jl [hunk 3]
```diff
@@ -106,17 +101,6 @@ end
 
 @adjoint collect(x::Array) = collect(x), Δ -> (Δ,)
 
-@adjoint fill(x::Real, dims...) = fill(x, dims...), Δ->(sum(Δ), map(_->nothing, dims)...)
-
-@adjoint function circshift(A, shifts)
-  circshift(A, shifts), Δ -> (circshift(Δ, map(-, shifts)), nothing)
-end
-
-@adjoint function reverse(x::AbstractArray, args...; kwargs...)
```
Comment (DhairyaLGandhi) [inline src/lib/array.jl:115]: Ditto here - this is needed for bidirectional recurrent cells

**Review by DhairyaLGandhi — APPROVED** [review]
Lgtm, couple of quick qs before merging

File: src/lib/array.jl (hunk 3 above)
Comment (willtebbutt) [reply src/lib/array.jl:115]: Added in 1.5: https://github.com/JuliaDiff/ChainRules.jl/blob/7593339839832ffb7ac83401918390ffe9d4eb42/src/rulesets/Base/array.jl#L314

The tests appear to pass -- are there not tests for `reverse` in Zygote?

File: src/lib/array.jl (hunk 2 above)
Comment (willtebbutt) [reply src/lib/array.jl:21]: Added in 1.5: https://github.com/JuliaDiff/ChainRules.jl/commit/7593339839832ffb7ac83401918390ffe9d4eb42#diff-7a31ac47520a0ad235f15cead3f4c48fcb595dc7f2421ebbee8ffce23e2510f5R7

File: src/lib/array.jl (hunk 3 above)
Comment (DhairyaLGandhi) [reply src/lib/array.jl:115]: Ah we are still testing, excellent?

**Pull request merged**